### PR TITLE
fix for escape key processing

### PR DESCRIPTION
**Pull request by willmcgugan**
FIxes escape key processing. Previously moving the mouse and hitting escape repeatedly could put the parser in a state where it wasn't producing any keys.

Also fixes an error that broke Windows.

Fixes https://github.com/Textualize/textual/issues/387

### CODE REVIEW

File: src/textual/_xterm_parser.py [hunk 1]
```diff
@@ -83,10 +88,23 @@ def parse(self, on_token: TokenCallback) -> Generator[Awaitable, str, None]:
...
+            if character == ESC:
+                # Could be the escape key was pressed OR the start of an escape sequence
                 sequence: str = character
+                peek_buffer = yield self.peek_buffer()
+                if not peek_buffer:
+                    # An escape arrived without any following characters
+                    on_token(events.Key(self.sender, key=ESC))
+                    continue
+                if peek_buffer and peek_buffer[0] == ESC:
+                    # There is an escape in the buffer, so ESC ESC has arrived
```
Comment (DrBenton) [inline src/textual/_xterm_parser.py:100]: :ok_hand:

**Review by DrBenton — DISMISSED** [review]
That must have been some nice investigation work! :detective: :slightly_smiling_face:

File: src/textual/_parser.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def __init__(self, sep: str, max_bytes: int | None = None) -> None:
         self.max_bytes = max_bytes
 
 
-class PeekBuffer(Awaitable):
+class _PeekBuffer(Awaitable):
```
Comment (willmcgugan) [inline src/textual/_parser.py:50]: Not sure why this wasn't marked as private.

File: tests/test_parser.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                if not data:
+                    break
+                on_token(data)
+
+    test_data = "Where there is a Will there is a way!"
+
+    for read_size in range(1, len(test_data) + 1):
+        test_parser = TestParser(read_size)
+        for size in range(1, len(test_data) + 1):
+            test_parser = TestParser(read_size)
```
[the remark this answers is not in the document]
Comment (willmcgugan) [reply tests/test_parser.py:51]: The first one needs to be removed.

**Pull request merged**